### DISCUSSION TITLE
Nuxt 3: filenames chunk not working?

### DISCUSSION THREAD

**Discussion by djimnz** [nuxt/nuxt#16248] [Questions] [answered]
Hello everyone,

I tried to change my chunk names by editing my nuxt config with:

``` 
{
    ...
    build: {
        filenames: {
             chunk: ({ isDev }) => (isDev ? '[name].[contenthash].js' : '[id].[contenthash].js')
        }
    }
    ...
}
```

But after my Netlify deploys, the chunks are still using the default pattern of: componentName.hash.js, such as for example index.d1b99a32.js

Did I forgot something?

Thank you!

**Comment by TheAlexLichter** [comment, marked as answer]
Hey! `build.filenames` is a Nuxt2 param.

Check out https://github.com/nuxt/framework/pull/6505

```
import { defineNuxtConfig } from 'nuxt'

export default defineNuxtConfig({
  vite: {
    build: {
      rollupOptions: {
        output: {
          chunkFileNames: '_nuxt/[hash].mjs'
        }
      }
    }
  }
})
```

**Reply by djimnz** [reply]
Hi @manniL 

Thank you so much! That was the solution. Do you have any particular preferences about .mjs over .js?

I'll check to do the same for my .css now and also, try to fix the queuing issue I have due to my CSS loading (I guess. My components downloads are delayed due to the app css files).
<img width="1349" alt="Screen Shot 2022-09-30 at 10 52 29 AM" src="https://user-images.githubusercontent.com/949997/193173273-fd31652a-d530-4a9e-97a2-49b5bcaa9458.png">

Have a great day and thank you again :)

**Reply by TheAlexLichter** [reply]
Hey!
No no, js is fine, just copied the example from the issue above while being lazy 🙊

**Reply by djimnz** [reply]
Final configuration solution for curious fellows: https://github.com/nuxt/framework/discussions/7911#discussioncomment-3776829

**Comment by djimnz** [comment]
For a anyone looking into renaming their chunks and assets names, here what to do, add the following in your `nuxt.config.js` `defineNuxtConfig`:

```
vite: {
        build: {
            rollupOptions: {
                output: {
                    chunkFileNames: '_nuxt/[hash].js', // can be .mjs or .js
                    assetFileNames: '_nuxt/[hash][extname]'
                }
            }
        }
    },
```

Official documentation:
https://rollupjs.org/guide/en/#outputchunkfilenames
https://rollupjs.org/guide/en/#outputassetfilenames

Hope that will save you time.

(cc @manniL)

**Reply by esseifforte** [reply]
wow, cool!
I have tested assetFileNames: '_nuxt/[hash].[extname]'
it works fine!!
thanks
